lib/cylc/config.py

```diff
@@ -1649,7 +1649,7 @@ def generate_taskdefs(self, orig_expr, left_nodes, right, seq):
             # Record custom message outputs.
             for item in self.cfg['runtime'][name]['outputs'].items():
                 if item not in taskdef.outputs:
```

Comment:
Don't think the `if` is required now (as a set items are unique), although there's a contained message offset check following the `add`.

Reply:
Good point - I didn't feel comfortable wiping out the obsolte task message offset check below.

Reply:
@dwsutherland - the if is still needed, this is where items are added to the set.  The loop is over dict items, not set items (i.e. `self.cfg['runtime'][name]['outputs']` is a dict).

@trwhitcomb - however, in the Cylc 8 PR for this you can remove the checkout of old-style message offsets (we are cleaning out pre-Cylc-7 cruft).

Reply:
> @trwhitcomb - however, in the Cylc 8 PR for this you can remove the checkout of old-style message offsets (we are cleaning out pre-Cylc-7 cruft).

Actually, don't bother - that's technically off-topic for this change. I'll make a separate fix for that.

Reply:
Sorry, misunderstood @dwsutherland 's point above. Yes you can just add an item to a set without testing for membership, so the `if` in line 1651 not technically required anymore.  However, maybe it's not necessarily faster without the pre-add membership test: https://stackoverflow.com/questions/29928634/should-i-check-if-an-item-is-already-in-a-set-before-adding-it